model-providers/ollama/runtime/src/main/java/io/quarkiverse/langchain4j/ollama/OllamaRestApi.java

```diff
@@ -85,40 +86,41 @@ class OllamaRestApiReaderInterceptor implements ReaderInterceptor {
         public Object aroundReadFrom(ReaderInterceptorContext context) throws IOException, WebApplicationException {
             try {
                 return context.proceed();
-            } catch (ClientWebApplicationException e) {
-                Throwable cause = e.getCause();
-                if ((cause instanceof JsonParseException) || (cause instanceof MismatchedInputException)) {
-                    if (e.getResponse().getStatus() == 200) {
-                        Object invokedMethod = context.getProperty("org.eclipse.microprofile.rest.client.invokedMethod");
-                        if ((invokedMethod != null) && invokedMethod.toString().contains("OllamaRestApi.streamingChat")) {
-                            InputStream is = context.getInputStream();
-                            if (is instanceof ByteArrayInputStream bis) {
-                                bis.reset();
-                                String chunk = new String(bis.readAllBytes());
-                                final var ctx = Vertx.currentContext();
-                                if (ctx == null) {
-                                    throw e;
-                                }
+            } catch (ClientWebApplicationException | ProcessingException e) {
+                if ((e instanceof ProcessingException pe && pe.getCause() instanceof MismatchedInputException) ||
+                        (e instanceof WebApplicationException wae
+                                && ((wae.getCause() instanceof JsonParseException && wae.getResponse().getStatus() == 200) ||
+                                        (wae.getCause() instanceof MismatchedInputException
+                                                && wae.getResponse().getStatus() == 400)))) {
```

Comment:
Actually we should probably have a comment on why this is needed (i.e. the change in Quarkus).

Reply:
Yes, i will add a comment

Reply:
🙏🏽